cli/run_util.go

```diff
@@ -540,14 +583,66 @@ func (r WorkflowRunner) activateAndRunSteps(
 	secrets []envmanModels.EnvironmentItemModel,
 	isLastWorkflow bool,
 	tracker analytics.Tracker,
-	workflowIDProperties coreanalytics.Properties) models.BuildRunResultsModel {
+	workflowIDProperties coreanalytics.Properties,
+	workflowID string,
+) models.BuildRunResultsModel {
 	log.Debug("[BITRISE_CLI] - Activating and running steps")
 
 	if len(workflow.Steps) == 0 {
 		log.Warnf("%s workflow has no steps to run, moving on to the next workflow...", workflow.Title)
 		return buildRunResults
 	}
 
+	err := tools.EnvmanInit(configs.InputEnvstorePath, true)
+	if err != nil {
+		log.Debugf("Couldn't initialize envman.")
+	}
+	err = tools.EnvmanAddEnvs(configs.InputEnvstorePath, *environments)
+	if err != nil {
+		log.Debugf("Couldn't add envs.")
+	}
+	envList, err := tools.EnvmanReadEnvList(configs.InputEnvstorePath)
+	if err != nil {
+		log.Debugf("Couldn't read envs from envman.")
+	}
+
+	serviceContainers, err := r.dockerManager.StartServiceContainers(workflow.Services, workflowID, envList)
+	if err != nil {
+		log.Errorf("❌ Some services failed to start properly!")
```

Comment:
Don't want to block the PR, but we should provide better errors than "some" services have "some" problems, it's going to be really hard to debug this way.

Reply:
We have proper logging in place if you take a look in the container manager. This is the very top where the error will bubble up, but the causes will be logged as much as possible before this happens.

Reply:
The reason we say "some" here, instead of the concrete names is because it would require us to also return a state of the services (as we start all of them to optimize boot time before we return from here). But inside this method each service failure will be logged individually.

Reply:
True!